### PR TITLE
Bid ask strategy

### PR DESCRIPTION
**Pull request by yperfanov**
## Summary
Orderbook available in the strategy.

Solve the issue: #1874

## What's new?
1. Implemented freqtrade/data/dataprovider.py ordebook() method.
2. Added commented example how to use it to retrieve best bid and ask in user_data/strategies/test_strategy.py.

### CODE REVIEW

**Comment by xmatthias** [comment]
Great :+1: 

It would be great to have this documented [here](
https://github.com/freqtrade/freqtrade/blob/develop/docs/strategy-customization.md#additional-data-dataprovider)

Note that in the strategy, i would always advise to check if `self.dp` is available.

File: user_data/strategies/test_strategy.py [hunk 1]
```diff
@@ -253,6 +253,15 @@ def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame
         dataframe['ha_low'] = heikinashi['low']
         """
 
+	# Retrieve best bid and best ask
+        # ------------------------------------
+        """
+        ob = self.dp.orderbook(metadata['pair'], 1)
+        print(ob)
```
Comment (hroff-1902) [inline user_data/strategies/test_strategy.py]: Super, thanks!

This print() should be commented out for merging to the freqtrade codebase.
Or removed.

**Comment by xmatthias** [comment]
yes but the existing section 5 lines above the new one also uses this format. Also it's a comment / documentation section, so i wouldn't see this too critical.

**Comment by hroff-1902** [comment]
> yes but the existing section 5(ish) lines above the new one also uses this format. Also it's a comment / documentation section, so i wouldn't see this too critical.

Sure, LGTM 👍 
(How to change the 'type' of the review comment above from 'request changes' to 'approve'?)

**Comment by xmatthias** [comment]
i think you can approve when you go to the files-section again ... but you can't change the previous review.

**Comment by hroff-1902** [comment]
> i think you can approve when you go to the files-section again ... but you can't change the previous review.

How's life complicated... :sarcasm

**Pull request merged**